features/integration/simulate.feature

```diff
@@ -0,0 +1,84 @@
+Feature: Simulating transactions
+  Background:
+    Given an algod v2 client
+    And a kmd client
+    And wallet information
+    And suggested transaction parameters from the algod v2 client
+    And I create a new transient account and fund it with 100000000 microalgos.
+    And I make a transaction signer for the transient account.
+
+  @simulate
+  Scenario Outline: Simulating successful payment transactions
+    Given default transaction with parameters <amt> "<note>"
+    When I get the private key
+    And I sign the transaction with the private key
+    And I simulate the transaction
+    Then the simulation should succeed without any failure message
```

Comment:
> Should we check that this didn't cost anything? (I'm leaning to "NO" even though I asked the question. But the next one, might be more of a "YES")

I think this would be closer to a ledger test (checking if the ledger accounts have not been modified by the simulation)

> Should we check that it fails for an over-spend?

Sure, I can add this test case without affecting any steps in downstream SDKs.

Reply:
Addressed in 90fdb79ac98c2340c04d86fc7e6167f9af8b2b25